homeassistant/components/camera/synology.py

```diff
@@ -0,0 +1,131 @@
+"""
+Support for Synology Surveillance Station Cameras.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.synology/
+"""
+import logging
+
+import voluptuous as vol
+
+import requests
+
+from homeassistant.const import (
+    CONF_NAME, CONF_USERNAME, CONF_PASSWORD,
+    CONF_URL, CONF_CAMERA_NAME, CONF_STREAM_ID)
+from homeassistant.components.camera import (Camera, PLATFORM_SCHEMA)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Synology Camera'
+DEFAULT_STREAM_ID = '0'
+TIMEOUT = 10
+
+# pylint: disable=line-too-long
+MJPEG_URL = '{0}/webapi/SurveillanceStation/streaming.cgi?api=\
+SYNO.SurveillanceStation.Streaming&method=LiveStream&version=1&cameraId={1}'
+STILL_IMAGE_URL = '{0}{1}'
+LOGIN_URL = '{0}/webapi/auth.cgi?api=SYNO.API.Auth&method=\
+Login&version=2&account={1}&passwd={2}&session=SurveillanceStation&format=sid'
+URL = '{0}/webapi/entry.cgi?\
+api=SYNO.SurveillanceStation.Camera&method=List&version=1'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_CAMERA_NAME): cv.string,

```

Comment:
Since you can fetch all camera's, remove this and add an optional CONF_WHITELIST param which is a list of cameras that are approved. If omitted, all cameras will be added.